lua/cfc_weapon_lockouts/server/sv_locker.lua

```diff
@@ -0,0 +1,216 @@
+    if lostWeapon or ( lostWeapon == nil and not ply:HasWeapon( weaponClass ) ) then
+        ply.weaponLockoutWeapons = ply.weaponLockoutWeapons or {}
+        ply.weaponLockoutWeapons[weaponClass] = nil
+    end
+
+    lockouts[weaponClass] = true
+    wep.weaponLockoutIsLocked = true
+
+    net.Start( "CFC_WeaponLockouts_LockWeapon" )
+    net.WriteTable( {
+        ply = ply,
+        wep = wep,
+        class = weaponClass
+    } )
+    net.Broadcast()
+
+    timer.Create( "CFC_WeaponLockouts_Unlock_" .. ply:SteamID() .. "_" .. weaponClass, CFCWeaponLockouts.LOCKOUT_TIME, 1, function()
+        if not IsValid( ply ) then return end
+
+        lockouts[weaponClass] = nil
+        wep.weaponLockoutIsLocked = nil
+
+        net.Start( "CFC_WeaponLockouts_UnlockWeapon" )
+        net.WriteTable( {
+            ply = ply,
+            wep = wep,
+            class = weaponClass
+        } )
+        net.Broadcast()
+    end )
+end
+
+local function warnPlayer( identifier, ply, warns )
+    if not IsValid( ply ) or table.IsEmpty( warns ) then return end
+
+    local lockedCount = warns.lockedCount or 0
+    local unlockedCount = warns.unlockedCount or 0
+
+    if lockedCount <= 0 then return end
+
+    local msg = "temporarily locked! You will be able to fire"
+    local lineLength = 0
+    local maxLength = 50
+    local maxAppend = "and more"
+
+    if CFCNotifications then
+        msg = "temporarily locked!\nYou will be able to fire"
+        maxLength = 100
+    end
+
+    if unlockedCount > 0 then
+        if lockedCount > 1 then
+            local append = "They are "
+            lineLength = lineLength + append:len()
+            msg = "Some of those weapons are " .. msg .. " them after a brief moment.\n" .. append
+        else
+            local append = "It is "
+            lineLength = lineLength + append:len()
+            msg = "One of those weapons is " .. msg .. " it after a brief moment.\n" .. append
+        end
+    else
+        if lockedCount > 1 then
+            local append = "They are "
+            lineLength = lineLength + append:len()
+            msg = "Those weapons are " .. msg .. " them after a brief moment.\n" .. append
+        else
+            local append = "It is "
+            lineLength = lineLength + append:len()
+            msg = "That weapon is " .. msg .. " it after a brief moment.\n" .. append
+        end
+    end
+
+    for class, status in pairs( warns ) do
+        if status == "locked" then
+            local newLength = lineLength + class:len()
+
+            if newLength <= maxLength then
+                if newLength + maxAppend:len() < maxLength then
+                    msg = msg .. class .. " "
+                    lineLength = newLength + 1
+                else
+                    msg = msg .. maxAppend
+                    break
+                end
+            else
+                msg = msg .. maxAppend
+                break
+            end
+        end
+    end
```

Comment:
This is all just string formatting. I think it should be pulled out into its own helper function - it's not necessarily relevant to this function's responsibility